client/src/main/scala/com.gu.contentapi.client/model/ContentApiError.scala

```diff
@@ -0,0 +1,15 @@
+package com.gu.contentapi.client.model
+
+import com.gu.contentapi.client.model.v1.ErrorResponse
+import com.gu.contentapi.client.thrift.ThriftDeserializer
+import scala.util.Try
+
+case class ContentApiError(httpStatus: Int, httpMessage: String, errorResponse: Option[ErrorResponse] = None) extends Exception(httpMessage)
+
+object ContentApiError {
+  def apply(response: HttpResponse): ContentApiError = {
+    val errorResponse = Try(ThriftDeserializer.deserialize(response.body, ErrorResponse)).toOption
+    ContentApiError(response.statusCode, response.statusMessage, errorResponse)
+  }
```

Comment:
If I may ask, what's the difference between having this `apply` method in the companion object and adding it as a secondary constructor to the class?

Reply:
Constructors do not seem very idiomatic in Scala.